src/cli/cmd/login/mod.rs

```diff
@@ -0,0 +1,95 @@
+use tokio::io::AsyncWriteExt;
+
+use super::CommandExecute;
+
+/// Login to FlakeHub in order to allow authenticated fetching of flakes.
+#[derive(Debug, Parser)]
+pub(crate) struct LoginSubcommand {
+    #[clap(from_global)]
+    api_addr: url::Url,
+
+    #[clap(from_global)]
+    frontend_addr: url::Url,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for LoginSubcommand {
+    async fn execute(self) -> color_eyre::Result<ExitCode> {
+        self.manual_login().await?;
+
+        Ok(ExitCode::SUCCESS)
+    }
+}
+
+impl LoginSubcommand {
+    async fn manual_login(&self) -> color_eyre::Result<()> {
+        // FIXME: this should really be the frontend, but the frontend doesn't have a /login path
+        // yet...
+        let mut login_url = self.api_addr.clone();
+        login_url.set_path("login");
+        login_url.set_query(Some("redirect=/token/create"));
+
+        println!("Login to FlakeHub: {}", login_url);
+        println!("And then follow the prompts below:");
+        println!();
+
+        let token = crate::cli::cmd::init::prompt::Prompt::maybe_string("Paste your token here:");
+        let token = match token {
+            Some(token) => {
+                // FIXME: validate that the token is valid?
+                // or at least validate that it's a... jwt at all lol
+                token
+            }
+            None => {
+                tracing::error!("Missing token.");
+                std::process::exit(1);
+            }
+        };
+
+        let xdg = xdg::BaseDirectories::new()?;
+
+        // $XDG_CONFIG_HOME/nix/nix.conf; basically ~/.config/nix/nix.conf
+        let nix_config_path = xdg.place_config_file("nix/nix.conf")?;
+        // $XDG_DATA_HOME/fh/netrc; basically $/.local/share/fh/netrc
+        let netrc_path = xdg.place_data_file("fh/netrc")?;
+
+        let nix_config_addition = format!("netrc-file = {}\n", netrc_path.display());
+        let netrc_contents = format!(
+            "\
+            machine {frontend_host} login FIXME password {token}\n\
+            machine {backend_host} login FIXME password {token}\n\
+            ",
+            frontend_host = self.frontend_addr.host_str().expect("FIXME"),
+            backend_host = self.api_addr.host_str().expect("FIXME"),
+        );
+
+        // NOTE: Keep an eye on any movement in the following issues / PRs. Them being resolved
+        // means we may be able to ditch setting `netrc-file` in favor of `access-tokens`.
+        // https://github.com/NixOS/nix/pull/9145 ("WIP: Support access-tokens for fetching tarballs from private sources")
+        // https://github.com/NixOS/nix/issues/8635 ("Credentials provider support for builtins.fetch*")
+        // https://github.com/NixOS/nix/issues/8439 ("--access-tokens option does nothing")
+        tokio::fs::write(netrc_path, netrc_contents).await?;
+
+        if crate::cli::cmd::init::prompt::Prompt::bool(&format!(
+            "May I add `{nix_config_addition}` to {}?",
```

Comment:
Trim the \n's in this prompt please:

```
> May I add `netrc-file = /Users/grahamc/.local/share/fh/netrc
` to /Users/grahamc/.config/nix/nix.conf? y
```

Reply:
Is there a good reason to ask about this, vs. just saying we did it?

Reply:
Yes, because it's not an append-field, so if the user already has a `netrc-file`, it will override all other `netrc-file` options they may have set. If they suddenly stop being able to pull e.g. corp repos because we didn't ask if we could, they might run away and not look back.